### DISCUSSION TITLE
I'm having trouble updating an element in a block

### DISCUSSION THREAD

**Discussion by atorres868** [yoopta-editor/Yoopta-Editor#191] [Q&A] [answered]
Hi!
I'm using version 4.6.0, I have tried to update the text of an element using the editor.updateBlock function, the "change" event tells me through the console that the change has been made, but the editor interface does not reflect the same , I share the code I'm using.

```typescript
const editor = useMemo(() => createYooptaEditor(), [])

const updateBlock = useCallback((text: string, focusBlockId: string) => {
    
    const block = editor.children[focusBlockId]
	    	    
    if (block) {
       
        editor.updateBlock(focusBlockId, {
          type: 'Paragraph',
          value: [
            {
              ...block.value[0],
              type: 'paragraph',
              props: { nodeType: 'block' },
              children: [{ text }],
            },
          ],
        }) 
       
       }
    },
    [editor]
  )
  
const newTextForBlock = 'Testing....'   
const currentBlock = findPluginBlockBySelectionPath(editor, {
      at: editor.selection,
})

const focusBlockId = currentBlock.id

updateBlock(newTextForBlock, focusBlockId) 
``` 
This text should be replaced with "Testing...."

![image](https://github.com/Darginec05/Yoopta-Editor/assets/45054828/971def1f-e649-4a89-bf6b-a94a0278dfd3)

This is what the change event shows me:

![Untitled](https://github.com/Darginec05/Yoopta-Editor/assets/45054828/cd179618-25fb-4fa0-8d93-7b0d7dac2254)

**Comment by Darginec05** [comment]
You mean you want to programmatically change the text in the block, right?

**Reply by atorres868** [reply]
That's correct, because I am getting that text from an API and I want to insert it there

**Reply by Darginec05** [reply]
@atorres868 got it. 
So, I need task to add new API method (updateElementText) for Elements entity

**Reply by Darginec05** [reply, marked as answer]
Hi @atorres868 :)
Task is done and new API available in v4.6.1.
I added to new methods: 
 - `Blocks.getBlock(editor: YooEditor, options: { id?: string, at?: string })`
 - `Elements.updateElementText(editor: YooEditor, text: string, options?: { blockId?: string, focus?: boolean  })`
